### ISSUE TITLE
Is there a performance impact when required across multiple files?

### ISSUE THREAD

**Issue by yattias** [lorenwest/node-config#365]
Over instantiation of file descriptors in unix system can result in lower performance.

That being said, will ``node-config`` attempt to access the specified json config file in situations where it is included in multiple node files? (user_controller, post_controller, ...). Or instead, does node/node-config bootstrap with the file in memory and prevent the constant allocation of file descriptors?

**Comment by lorenwest** [comment]
`node-config` performs all file I/O on first invocation, generally on program startup.

It attempts to open many files in a pre-determined order. If the file exists it reads the file and closes the file descriptor - one at a time.

The file is loaded, parsed, and merged with other configurations into a single configuration object in memory. The object is made immutable, and during the lifetime of the app, config access is made to this immutable in-memory object.

In short, the file descriptor use of `node-config` is extremely light and only on program load.

**Comment by yattias** [comment]
Got it, thanks for the detailed explanation!
